### PR TITLE
[DOCS] Metric component examples

### LINKED ISSUE #1383: [DOCS] Add more example for metric
Adding more example for metric component

### PR DESCRIPTION
**Pull request by namnguyen20999**
Example for Metric component

### CODE REVIEW

File: doc/gui/examples/extensions/metric-shape.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+default_value = 50
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/extensions/metric-shape.py]: remove this ?

Comment (FredLL-Avaiga) [reply doc/gui/examples/extensions/metric-shape.py]: show that delta can be bound ?

File: doc/gui/examples/extensions/metric-simple.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+# Initial value
+value = 90
+
+page = """
+<|{value}|metric|show_value=true|min=50|max=150|delta=20|threshold=100|format=%d km/h|delta_format=%dkm/h|width=800px|height=500px|>
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/extensions/metric-simple.py]: show threshold can be bound

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: doc/gui/examples/extensions/metric-shape.py
Comment (FabienLelaquais) [inline doc/gui/examples/extensions/metric-shape.py:1]: The target path should be `doc/gui/examples/controls/metric-type.py`.
The 'shape' is a notion that is not reflected in property names.

File: doc/gui/examples/extensions/metric-shape.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# Initial value
+value = 50
+
+page = """
+<|{value}|metric|type=linear|delta=-20|format=%%|delta_format=%%|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/extensions/metric-shape.py]: `format` and `delta_format` sould be dropped: you are showing the property 'type'.

File: doc/gui/examples/extensions/metric-simple.py
Comment (FabienLelaquais) [inline doc/gui/examples/extensions/metric-simple.py:1]: Path should be `doc/gui/examples/controls/metric-simple.py`

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
The files path must be changed,

File: doc/gui/examples/extensions/metric-simple.py (hunk 2 above)
Comment (FabienLelaquais) [inline doc/gui/examples/extensions/metric-simple.py]: - This line is too long for the linter.
   Split it by concatenating string fragments.
- Remove min (default to 0, perfect for a speed).
- Remove width and height unless they bring something useful.

File: doc/gui/examples/extensions/metric-shape.py (hunk 1 above)
Comment (namnguyen20999) [reply doc/gui/examples/extensions/metric-shape.py]: resolved

File: doc/gui/examples/extensions/metric-simple.py (hunk 2 above)
Comment (namnguyen20999) [reply doc/gui/examples/extensions/metric-simple.py]: resolved

File: doc/gui/examples/extensions/metric-shape.py (hunk 3 above)
Comment (namnguyen20999) [reply doc/gui/examples/extensions/metric-shape.py]: resolved

File: doc/gui/examples/controls/metric-no-show-value.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+page = """
+<|90|metric|show_value=false|>
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/controls/metric-no-show-value.py]: Don't show_value is nicer

File: doc/gui/examples/controls/metric-simple.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+delta_value = 20
+min_value = 50
+max_value = 150
+threshold = 100
+
+page = """
+<|90|metric|show_value=true|min={min_value}|max={max_value}|delta={delta_value}|threshold={threshold}|format=%d km/h|delta_format=%dkm/h|width=800px|height=500px|>
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/controls/metric-simple.py]: No need for Show_value= true (default)

File: doc/gui/examples/controls/metric-layout.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+# Layout reference can be found in the documentation: https://plotly.com/python/reference/layout/
+
+custom_layout = {
+    "width": "1000",
+    "height": "500",
+}
+
+page = """
+<|30|metric|layout={custom_layout}|>
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/controls/metric-layout.py]: Can use width and height directly
Can you show a differentiating use case

File: doc/gui/examples/controls/metric-type.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+# specific language governing permissions and limitations under the License.
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+page = """
+<|50|metric|type=linear|>
```
Comment (FredLL-Avaiga) [inline doc/gui/examples/controls/metric-type.py]: Use a variable for 50

**Review by FredLL-Avaiga — DISMISSED** [review]

File: doc/gui/examples/controls/metric-layout.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+# -----------------------------------------------------------------------------------------
+# To execute this script, make sure that the taipy-gui package is installed in your
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+# Layout reference can be found in the documentation: https://plotly.com/python/reference/layout/
+
+custom_layout = {
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/metric-layout.py]:
```suggestion
layout = {
```

File: doc/gui/examples/controls/metric-layout.py (hunk 6 above)
Comment (FabienLelaquais) [reply doc/gui/examples/controls/metric-layout.py]: If you want an example of using the layout, I would want a more interesting example than just width or height. What other impact can `layout` have on a gauge?
```suggestion
<|30|metric|layout={layout}|>
```

File: doc/gui/examples/controls/metric-min-max-range.py
Comment (FabienLelaquais) [inline doc/gui/examples/controls/metric-min-max-range.py:1]: -> metric-range.py

File: doc/gui/examples/controls/metric-min-max-range.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+# Python environment and run:
+#     python <script>
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+min_value = 50
+max_value = 150
+
+page = """
+<|90|metric|min={min_value}|max={max_value}|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/metric-min-max-range.py]: I think a more common use case is where `value` is a bound value, and `min` and `max` not so much.

File: doc/gui/examples/controls/metric-no-show-value.py
Comment (FabienLelaquais) [inline doc/gui/examples/controls/metric-no-show-value.py:1]: -> metric-hide-value.py

File: doc/gui/examples/controls/metric-no-show-value.py (hunk 4 above)
Comment (FabienLelaquais) [reply doc/gui/examples/controls/metric-no-show-value.py]: Actually: `don't show_value`

File: doc/gui/examples/controls/metric-simple.py (hunk 5 above)
Comment (FabienLelaquais) [reply doc/gui/examples/controls/metric-simple.py]: Using formats in the simple case is too much. Please drop formats as well.
That is taken care of in the `-format` example.

File: doc/gui/examples/controls/metric-value-format.py
Comment (FabienLelaquais) [inline doc/gui/examples/controls/metric-value-format.py:1]: -> metric-format.py

File: doc/gui/examples/controls/metric-value-format.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+# -----------------------------------------------------------------------------------------
+from taipy.gui import Gui
+
+delta_value = 20
+
+# format & delta_format are used to format the value and delta value respectively.
+# They use the printf syntax.
+
+page = """
+<|90|metric|delta={delta_value}|format=%d km/h|delta_format=%dkm/h|>
```
Comment (FabienLelaquais) [inline doc/gui/examples/controls/metric-value-format.py]:
```suggestion
<|90|metric|delta={delta_value}|format=%d km/h|delta_format=%d km/h|>
```

**Review by FabienLelaquais — APPROVED** [review]
Mostly minor things.

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Please address the renaming topics.

File: doc/gui/examples/controls/metric-no-show-value.py (hunk 4 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-no-show-value.py]: resolved

File: doc/gui/examples/controls/metric-simple.py (hunk 5 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-simple.py]: resolved

File: doc/gui/examples/extensions/metric-simple.py (hunk 2 above)
Comment (namnguyen20999) [reply doc/gui/examples/extensions/metric-simple.py]: resolved

File: doc/gui/examples/controls/metric-value-format.py (hunk 10 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-value-format.py]: resolved

File: doc/gui/examples/controls/metric-value-format.py
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-value-format.py:1]: changed

File: doc/gui/examples/controls/metric-no-show-value.py
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-no-show-value.py:1]: I'll keep it as don't-show-value as mentioned above

File: doc/gui/examples/controls/metric-min-max-range.py
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-min-max-range.py:1]: changed

File: doc/gui/examples/controls/metric-layout.py (hunk 8 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-layout.py]: changed

File: doc/gui/examples/controls/metric-type.py (hunk 7 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-type.py]: resolved

File: doc/gui/examples/controls/metric-layout.py (hunk 6 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-layout.py]: I've added the background color, margin & font size for the example

File: doc/gui/examples/controls/metric-min-max-range.py (hunk 9 above)
Comment (namnguyen20999) [reply doc/gui/examples/controls/metric-min-max-range.py]: resolved

**Review by FabienLelaquais — APPROVED** [review]
Neat!

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**